### PR TITLE
Basic routing (ip-forwarding) test

### PR DESCRIPTION
**Pull request by ahmkar94**

### CODE REVIEW

File: test/infamy/netns.py [hunk 1]
```diff
@@ -67,11 +67,21 @@ def runsh(self, script, *args, **kwargs):
...
-    def addip(self, addr, subnet=24):
+    def addroute(self, subnet_ip, intermediary, prefix_length=""):
+        if prefix_length:
+            print("Prefix length empty")
+            prefix_length=f"/{prefix_length}"
+        
+        self.runsh(f"""
+            set -ex
+            ip route add {subnet_ip}{prefix_length} via {intermediary}
+            """, check=True)
```
Comment (troglobit) [inline test/infamy/netns.py:78]: - Drop the `print()`, seems to be debug (that's also wrong?)
- Rename `subnet_ip` to just `subnet`
- Rename `intermediary` to `gateway` (or `nexthop`), according to principle of least surprise

File: test/infamy/netns.py [hunk 2]
```diff
@@ -85,4 +95,4 @@ def must_reach(self, daddr):
     def must_not_reach(self, daddr):
         res = self.ping(daddr)
         if res.returncode == 0:
-            raise Exception(res.stdout)
+            raise Exception(res.stdout)
```
Comment (troglobit) [inline test/infamy/netns.py:98]: Seems to have snuck in some random whitespace here?

File: test/case/ietf_interfaces/routing_basic.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            
+            ns0.must_reach("10.0.0.10")
+            ns1.must_reach("192.168.0.10")
+
+    with test.step("Configure target physical ports [disable routing]"):
+
+        config_target(target, tport0, tport1, False)
+
+    with test.step("Test <ip routing disabled"):
+
```
Comment (troglobit) [inline test/case/ietf_interfaces/routing_basic.py:85]: Seems to be missing the closing `>` used in previous step?

**Review by troglobit — CHANGES_REQUESTED** [review]
Some minor things to fix, otherwise it looks great!

File: test/infamy/netns.py (hunk 1 above)
Comment (ahmkar94) [reply test/infamy/netns.py:78]: Thnx :)

**Review by troglobit — APPROVED** [review]

**Pull request merged**